packages/aws-cdk-lib/aws-elasticache/lib/elasticache-cluster.ts

```diff
@@ -0,0 +1,139 @@
+import * as cdk from 'aws-cdk-lib';
+import * as elasticache from 'aws-cdk-lib/aws-elasticache';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { Construct } from 'constructs';
+
+export interface ElastiCacheClusterProps {
+  /**
+   * The VPC where the ElastiCache cluster will be deployed
+   */
+  readonly vpc: ec2.IVpc;
+  
+  /**
+   * The cache engine version to use
+   * @default - 6.x (Redis)
+   */
+  readonly engineVersion?: string;
+  
+  /**
+   * The node type for the ElastiCache cluster
+   * @default - cache.t3.micro
+   */
+  readonly nodeType?: string;
+  
+  /**
+   * Number of cache nodes in the cluster
+   * @default 1
+   */
+  readonly numCacheNodes?: number;
+  
+  /**
+   * The name of the cache parameter group to associate with this cache cluster
+   */
+  readonly parameterGroupFamily?: string;
+  
+  /**
+   * Security groups to attach to the cluster
+   * @default - A new security group is created
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+  
+  /**
+   * The weekly time range during which maintenance updates can occur.
+   * @default - No preference
+   */
+  readonly preferredMaintenanceWindow?: string;
```

Comment:
Since there is an allowed format for this property, in CDK we usually create a new Type that simplify it. For example this property can of a new Type which is:
```
interface DayTime {
     day Enum;
     time number; // we can add a validation that it is between 0-23
}
interface MaintenanceWindow {
      startMaintenanceWindow DayTime;
      endMaintenanceWindow DayTime; // we can add a validation that the end maintenance window is after the start
}
```